cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -45,7 +46,7 @@ import           Cardano.Api.Modes
 -- In order to make pipelining still possible we can explore the use of Selective Functors
 -- which would allow us to straddle both worlds.
 newtype LocalStateQueryExpr block point query r m a = LocalStateQueryExpr
-  { runLocalStateQueryExpr :: ContT (Net.Query.ClientStAcquired block point query m r) m a
+  { runLocalStateQueryExpr :: ReaderT NodeToClientVersion (ContT (Net.Query.ClientStAcquired block point query m r) m) a
```

Comment:
I was always suspect about using `ConT` here to get monadic local state query expressions. Can we not write our own `Monad` instance for which will potentially make it more pleasant to use? 

e.g `result <- ExceptT . fmap (join . first ShelleyQueryCmdAcquireFailure) $ ...` does not compose well.

Can we avoid the `fmap (join ...` by writing our own Monad instance? The type errors I experienced trying to get the expression to compile resulted in me abandoning trying to using it. Adding `ReaderT` here as things are now only serves to make this less usable imo.

Reply:
The `fmap (join ...` doesn't come from `ContT` and writing our own monad instance wouldn't fix it.

It comes from the fact that we are returning multiple error types.  It's one of my motivations for using `oops` which makes returning multiple error types seamless.